lib/core/src/client/preview/client_api.js

```diff
@@ -9,7 +9,11 @@ import subscriptionsStore from './subscriptions_store';
 
 export const defaultDecorateStory = (getStory, decorators) =>
   decorators.reduce(
-    (decorated, decorator) => context => decorator(() => decorated(context), context),
+    (decorated, decorator) => context =>
+      decorator(
+        (...args) => decorated({ ...context, arguments: [...context.arguments, ...args] }, ...args),
```

Comment:
Here's the feature implemented to merge in arguments passed into the decorator `fn`